rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/video.kt

```diff
@@ -30,4 +32,65 @@ internal fun DSPF.getDbFields(): List<DbField> {
     }
 
     return fields
+}
+
+internal fun List<FileDefinition>.toDSPF(): Map<String, DSPF>? {
+    val displayFiles = mutableMapOf<String, DSPF>()
+    val dspfConfig = MainExecutionContext.getConfiguration().dspfConfig
+
+    this.filter { it.fileType == FileType.VIDEO }.forEach {
+        if (dspfConfig != null) {
+            displayFiles[it.name] = dspfConfig.dspfProducer.invoke(it.name)
+        }
+        // if dspfConfig == null then display file fields have already been loaded into
+        // data definitions from .json metadata (reloadConfig is used as fallback)
+    }
+
+    // should I return null or an empty map?
+    return if (dspfConfig != null) displayFiles else null
+}
+
+/**
+ * Fields of specified record will be returned and updated with the latest
+ * value of the corrisponding data definition just before EXFMT starts.
+ */
+internal fun copyDataDefinitionsIntoRecordFields(interpreter: InterpreterCore, recordName: String): List<DSPFField> {
+    val fields = mutableListOf<DSPFField>()
+    val symbolTable = interpreter.getGlobalSymbolTable()
+    val displayFiles = interpreter.getStatus().displayFiles
+
+    displayFiles?.forEach { dspf ->
+        val record = dspf.value.records.first { it.name == recordName }
+        record.fields.forEach { field ->
+            field.value.primitive = symbolTable[field.name].asString().value
+            fields.add(field)
+        }
+    }
+
+    return fields
+}
+
+/**
+ * Fields edited during EXFMT will be available just after returning from it as response
+ * and used to update corresponding data definitions.
+ */
+internal fun copyRecordFieldsIntoDataDefinitions(interpreter: InterpreterCore, response: OnExfmtResponse) {
```

Comment:
This implementation will not be not so naive when response.values will be an instance of `Map<String, Value>`